### PR TITLE
Basic monitoring instructions

### PR DESCRIPTION
**Pull request by bion**

### CODE REVIEW

File: docs/it-manual/monitoring.md [hunk 1]
```diff
@@ -0,0 +1,719 @@
...
+The [CiviForm terraform deployment system](https://github.com/civiform/cloud-deploy-infra/) for AWS deploys the monitoring stack automatically. After deployment, user access to the Grafana dashboard and configuration of the dashboard need to be done manually.
+
+#### Configure access
+
+[AWS Managed Grafana](https://aws.amazon.com/grafana/) uses [AWS IAM Identity Center](https://aws.amazon.com/iam/identity-center/) for access management. **Note that this is a different service from [AWS IAM](https://aws.amazon.com/iam/) the accounts/user profiles in AWS IAM Identity Center are completely separate from accounts in AWS IAM.**
+
+1. Login to the AWS console and navigate to the IAM Identity Center service
+1. In the left nav, click "Users"
+1. For each user you'd like to grant access to viewing metrics, click the "Add user" button and follow the workflow
+1. Follow instructions [here](https://docs.aws.amazon.com/grafana/latest/userguide/AMG-manage-users-and-groups-AMG.html) for adding granting access for those users to your Grafana workspace
```
Comment (avrittrohwer) [inline docs/it-manual/monitoring.md:24]: I think it is worth calling out here that after adding a user, you'll need to separately make them an admin in the grafana workspace assigned users configuration

File: docs/it-manual/monitoring.md [hunk 2]
```diff
@@ -0,0 +1,719 @@
...
+
+#### Configure dashboard
+
+Once you have an Identity Center user with permissions to administer your Grafana workspace, it's time to configure the workspace dashboard. From the Grafana workspace page in the AWS console, click the link under "Grafana workspace URL". After signing in this will take you to your Grafana workspace. 
+
+To enable viewing metrics, add the Prometheus server as a data source for your Grafana workspace: 
+
+1. Click the AWS logo in the left nav
+1. Click "Amazon Managed Service for Prometheus"
+1. Add the data source for Prometheus
```
Comment (avrittrohwer) [inline docs/it-manual/monitoring.md:34]: I had to configure this from the AWS console and not grafana.  The only AWS logo in the sidebar linked to some docs.  How I added the data source:

1. Go to the workspace
2. Click on the 'Data sources' tab
3. Click on the 'Configure in Grafana' link on the 'Amazon Managed Service for Prometheus' row
4. Select the region where civiform is deployed
5. Select the 'avritt-dev-CiviForm_metrics' row
6. Select 'add 1 data source'

File: docs/it-manual/monitoring.md [hunk 3]
```diff
@@ -0,0 +1,719 @@
...
+
+With Prometheus connected as a Grafana workspace, panels can now be created in Grafana that display metrics from the CiviForm server. There are many metrics available, and many ways to display them. You can get started with the some basic metrics by importing a pre-built CiviForm dashboard:
+
+1. Hover over the '+' icon in the left nav
+1. Click the 'Import' option 
+1. Paste the following JSON into the "Import via panel JSON" and click "Load"
+
+```json
```
Comment (avrittrohwer) [inline docs/it-manual/monitoring.md:42]: Is there an easy way to add this file to the repo somewhere then make this a file download?

Comment (bion) [reply docs/it-manual/monitoring.md:42]: Probably. The reader needs to paste the actual text into the UI though, so making them download the file first adds extra steps for downloading, opening it in an editor. Thoughts?

File: docs/it-manual/monitoring.md [hunk 4]
```diff
@@ -0,0 +1,719 @@
...
+{
+  "annotations": {
```
Comment (avrittrohwer) [inline docs/it-manual/monitoring.md:44]: The graph import did not work (empty graphs), I suspect there are some UIDs that need to be just right

Comment (avrittrohwer) [reply docs/it-manual/monitoring.md:44]: Might be worth looking into https://grafana.com/docs/grafana/v9.0/dashboards/export-import/#making-a-dashboard-portable

File: docs/it-manual/monitoring.md [hunk 5]
```diff
@@ -0,0 +1,719 @@
+# Monitoring a CiviForm deployment
+
+CiviForm supports server monitoring via [Prometheus](https://prometheus.io/) metrics visualized in [Grafana](https://grafana.com/).
```
[the remark this answers is not in the document]
Comment (bion) [reply docs/it-manual/monitoring.md:3]: Done, added to the intro:

> These metrics are related to server status, things like latency and error rates, and do not contain sensitive data, such as who is applying to what programs.

File: docs/it-manual/monitoring.md [hunk 6]
```diff
@@ -0,0 +1,719 @@
...
+
+## Viewing metrics
+
+### AWS
+
+The [CiviForm terraform deployment system](https://github.com/civiform/cloud-deploy-infra/) for AWS deploys the monitoring stack automatically. After deployment, user access to the Grafana dashboard and configuration of the dashboard need to be done manually.
+
+#### Configure access
+
+[AWS Managed Grafana](https://aws.amazon.com/grafana/) uses [AWS IAM Identity Center](https://aws.amazon.com/iam/identity-center/) for access management. **Note that this is a different service from [AWS IAM](https://aws.amazon.com/iam/) the accounts/user profiles in AWS IAM Identity Center are completely separate from accounts in AWS IAM.**
```
[the remark this answers is not in the document]
Comment (bion) [reply docs/it-manual/monitoring.md:19]: Done

File: docs/it-manual/monitoring.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (bion) [reply docs/it-manual/monitoring.md:24]: Done

File: docs/it-manual/monitoring.md [hunk 7]
```diff
@@ -0,0 +1,719 @@
...
+To enable viewing metrics, add the Prometheus server as a data source for your Grafana workspace: 
+
+1. Click the AWS logo in the left nav
+1. Click "Amazon Managed Service for Prometheus"
+1. Add the data source for Prometheus
+
+With Prometheus connected as a Grafana workspace, panels can now be created in Grafana that display metrics from the CiviForm server. There are many metrics available, and many ways to display them. You can get started with the some basic metrics by importing a pre-built CiviForm dashboard:
+
+1. Hover over the '+' icon in the left nav
+1. Click the 'Import' option
```
[the remark this answers is not in the document]
Comment (bion) [reply docs/it-manual/monitoring.md:39]: Done

File: docs/it-manual/monitoring.md (hunk 3 above)
Comment (bion) [reply docs/it-manual/monitoring.md:42]: Switched to linking to the raw JSON content in a github gist

File: docs/it-manual/monitoring.md (hunk 4 above)
Comment (bion) [reply docs/it-manual/monitoring.md:44]: resolved

File: docs/it-manual/monitoring.md (hunk 1 above)
Comment (bion) [reply docs/it-manual/monitoring.md:24]: Done

File: docs/it-manual/monitoring.md (hunk 2 above)
Comment (bion) [reply docs/it-manual/monitoring.md:34]: Added

File: docs/it-manual/monitoring.md [hunk 8]
```diff
@@ -0,0 +1,56 @@
...
+- Server erors per minute (500 status code), split out by controller action
+- 50th percentile 5-minute trailing average request latency, split out by controller action 
+- 90th percentile 5-minute trailing average request latency, split out by controller action 
+- 99th percentile 5-minute trailing average request latency, split out by controller action 
+
+To import this pre-built dashboard: 
+
+1. Hover over the "+" icon in the left nav
+1. Click the "Import" option 
+1. Paste [the JSON here](followinghttps://gist.githubusercontent.com/bion/9c5e9293a887f4ea76f123908fb35ef7/raw/c179075f37026544124475ed46646861fda16a5f/civiform_grafana_dashboard.json) into the "Import via panel JSON" and click "Load"
```
Comment (avrittrohwer) [inline docs/it-manual/monitoring.md:55]: remove 'following' from the link

Comment (bion) [reply docs/it-manual/monitoring.md:55]: ah good catch, thanks!

**Pull request merged**